src/components/uikit/ProgressBar.tsx

```diff
@@ -0,0 +1,50 @@
+import { View } from 'react-native';
+
+import { styled } from '~styles';
+
+type Props = {
+  step: number;
+  totalSteps: number;
+  height?: number;
+};
+
+/**
+ * Renders a progress bar for a multi-step form.
+ *
+ * @param {Object} props - The component props.
+ * @param {number} props.step - The current step of the form.
+ * @param {number} props.totalSteps - The total number of steps in the form.
+ * @returns {JSX.Element} - The rendered component.
+ */
+export function ProgressBar({
+  step,
+  totalSteps,
+  height = 12,
+}: Props): JSX.Element {
+  let progress = (step / totalSteps) * 100;
```

Comment:
You can use `Math.min(Math.max(value, min), max)` to clamp the value without this if/else spaghetti 😛